lib/kafka_ex/network_client.ex

```diff
@@ -57,4 +59,15 @@ defmodule KafkaEx.NetworkClient do
       _ -> to_char_list(host)
     end
   end
+
+  defp default_socket_options do
+    default_options = [:binary, {:packet, 4}]
```

Comment:
Should be able to get rid of this - see my comment re: "options".  Just build the `%Socket{}` that you need.